dynawo/sources/Modeler/Common/DYNCompiler.cpp

```diff
@@ -84,21 +84,21 @@ Compiler::compile() {
   getDDB();
 
   unitDynamicModelsMap_ = dyd_->getUnitDynamicModelsMap();
-  map<string, shared_ptr<ModelDescription> > blackboxes = dyd_->getBlackBoxModelDescriptions();
-  map<string, shared_ptr<ModelDescription> >::const_iterator itbbm;
+  const boost::unordered_map<string, shared_ptr<ModelDescription> >& blackboxes = dyd_->getBlackBoxModelDescriptions();
+  boost::unordered_map<string, shared_ptr<ModelDescription> >::const_iterator itbbm;
```

Comment:
while we modify this part, please include the iterator into the for loop

Reply:
Also this means that the compilation will be done in a random manner doesn't this include non determinism in the logs? (it is OK in the compile log, but not in others)

Reply:
OK